File: Application/Backend/CodeNB.Model/Dtos/PurchaseOrderDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CodeNB.Model
{
    public class PurchaseOrderDto
    {

        public string PurchaseOrderNumber { get; set; } 
        public string? EmployeeFullName { get; set; } 
        public string? DepartmentName { get; set; } 
        public string? SupervisorFullName { get; set; } 
        public DateTime CreationDate { get; set; }
        public string PurchaseStatus { get; set; }
        public int EmployeeID { get; set; }
        public List<PurchaseOrderItemDto> Items { get; set; } = new();
        public byte[]? RecordVersion { get; set; }
        public decimal Subtotal { get; set; }
        public decimal TaxTotal { get; set; }
        public decimal GrandTotal { get; set; }
    }
}

File: Application/CodeNB-Web/src/app/core/models/purchase-order-dto.ts
import { PurchaseOrderItemDto } from "./purchase-order-item-dto";

export class PurchaseOrderDto {

    purchaseOrderNumber: string = '';
    employeeFullName?: string;
    departmentName?: string;
    supervisorFullName?: string;
    creationDate: Date = new Date();
    purchaseStatus: string = '';
    employeeID: number;
    recordVersion?: string;
    items: PurchaseOrderItemDto[] = [];
    subtotal: number = 0;
    taxTotal: number = 0;
    grandTotal: number = 0;
}
